File: GOF/Factory Method/java/src/maze/Room.java
package maze;

public class Room extends MapSite {
    public enum Direction {
        north, south, east, west
    }

    private int m_roomNumber;
    private MapSite[] m_sides = new MapSite[4];

    public Room(int n)
    {
        m_roomNumber = n;
    }

    public void setSide(Direction direction, MapSite site)
    {
        m_sides[direction.ordinal()] = site;
    }
}

File: GOF/抽象工厂AbstractFactory/cpp/maze/room.cpp
//
// Created by gaojian on 19-2-27.
//

#include "room.h"

Room::Room(int n)
	: m_roomNumber{n}
{}

void Room::setSide(Room::Direction direction, std::shared_ptr<MapSite> site)
{
	m_sides[direction] = std::move(site);
}